pkg/webhook/mutation.go

```diff
@@ -162,9 +166,47 @@ func (h *mutationHandler) Handle(ctx context.Context, req admission.Request) adm
 }
 
 func (h *mutationHandler) mutateRequest(ctx context.Context, req admission.Request) (admission.Response, error) {
-	// TODO: place mutation logic here
-	mutatedObject := req.Object
-	resp := admission.PatchResponseFromRaw(req.Object.Raw, mutatedObject.Raw)
+	okResp := admission.Response{
+		AdmissionResponse: admissionv1beta1.AdmissionResponse{
+			Allowed: true,
+			Result: &metav1.Status{
+				Code: int32(http.StatusOK),
+			},
+		},
+	}
+	ns := &corev1.Namespace{}
+	if err := h.client.Get(ctx, types.NamespacedName{Name: req.AdmissionRequest.Namespace}, ns); err != nil {
+		if !k8serrors.IsNotFound(err) {
+			log.Info("Namespace not found", "name", req.AdmissionRequest.Namespace)
+			return okResp, nil
+		}
+		// bypass cached client and ask api-server directly
+		err = h.reader.Get(ctx, types.NamespacedName{Name: req.AdmissionRequest.Namespace}, ns)
+		if err != nil {
+			log.Info("Namespace not found", "name", req.AdmissionRequest.Namespace)
+			return okResp, nil
+		}
+	}
+	obj := unstructured.Unstructured{}
+	err := obj.UnmarshalJSON(req.Object.Raw)
+	if err != nil {
+		log.Error(err, "Failed to unmarshal", "object", string(req.Object.Raw))
+		return okResp, nil
```

Comment:
I think this is the safer way to behave in case of errors, otherwise the mutating wh will block the cluster.
@maxsmythe thoughts?

Reply:
Maybe we need a switch that determines what to do in case of a failed mutation (in config? individually for each Assign?)

Reply:
I think we have two options:

* Reject the request outright
* Return a 500 error and let the validating webhook configuration decide the proper response

a 500 error is technically not correct HTTP, as this is a request error, not a server error. But stretching the convention seems fine to me if we can justify it.

TBH if we can't deserialize the object into an unstructured resource, then it's probably a malformed request and should be rejected, though there are cases where object would be nil... DELETE, for example. Not sure if DELETE is relevant for mutation.

Reply:
I think I'm okay with failing open by default here, since this is mutation, not validation.

Reply:
> Not sure if DELETE is relevant for mutation.

This is specified by the MutatingWebhookConfiguration. We currently have:

```
    operations:
    - CREATE
    - UPDATE

```
so in our case DELETE is not relevant.

Reply:
> I think I'm okay with failing open by default here, since this is mutation, not validation.

We have the webhook failurePolicy` failurePolicy: Ignore`, so the 500 should be ignored and the resource allowed in unmutated.

Reply:
I wouldn't necessarily base what's relevant on the MWH Config, as users are free to change that to look like whatever they want.

I do agree though, b/c I don't know why you'd need to mutate a DELETE operation.

+1 to the 500 approach. We can revisit this after tinkering with the prototype if need be.